Source/JavaScriptCore/jit/CCallHelpers.h

```diff
@@ -198,11 +198,17 @@ class CCallHelpers : public AssemblyHelpers {
 #if CPU(MIPS)
     template<unsigned NumCrossSources, unsigned NumberOfRegisters>
     ALWAYS_INLINE void setupStubCrossArgs(std::array<GPRReg, NumberOfRegisters> destinations, std::array<FPRReg, NumberOfRegisters> sources) {
-        for (unsigned i = 0; i < NumCrossSources; i++) {
-            GPRReg dest = destinations[i];
-            FPRReg source = sources[i];
+        if constexpr (!NumCrossSources) {
```

Comment:
Why not just `if constexpr (NumCrossSources)`?  Do we really need to handle this do nothing case?

Reply:
Not handling the else results in a warning on GCC 9.x (what the 32-bit JSC buildbots use). It's a clean build if we have UNUSED_PARAM at the entry or exit though, just thought that might have been more confusing to readers.

Reply:
butting in with a suggestion: reformulating the loop in terms of iterators instead of indices might avoid both the UNUSED_PARAM and the type-limits warning

Reply:
There's also `UNUSED_VARIABLE`.

Reply:
Right but the warnings are about unused parameters, not variables, so not sure how that's cleaner? One is defined to the other in any case.